### PR TITLE
Redesigns the Captain's gas mask into a Rebreather, makes it actually look good

### PR DESCRIPTION
**Pull request by DATA-xPUNGED**
## About The Pull Request
image
Captain's atmos mask is now a cool snazzy looking rebreather instead, with a dual-purpose slot for either a filter or narcotics. As it doesn't cover the eyes anymore, it doesn't have FOV restrictions and can also be put out of the way.
## Why It's Good For The Game

I've found myself going to sec roundstart to get a sechailer, solely because it looks way better than a default mask. Maybe this will make this item finally have some use? I'm also thinking of putting it in their internal box but it might be better as a completely one of a kind item.
## Changelog
:cl:
image: Nanotrasen has issued brand new, state of the art gas masks for their captains to use! Gone are the days of painted atmos masks.
balance: The new mask has many improvements over the old one. It doesn't cover the eyes, so it doesn't hinder FOV
/:cl:

Drafting until i make inhand sprites

### CODE REVIEW

**Comment by carlarctg** [comment]
Can't believe my shoddy resprite I made to fix an issue with the SWAT suit not being fireproof eventually blossomed into this beautiful design. Excellent job.

**Comment by carlarctg** [comment]
Nitpick: The object icon shows the mask as being pretty thin around the sides, but the onmob has the blue cloth(?) stretched across the whole face like a bandana. Feels inconsistent. (And personally speaking doesn't look captainy)

**Comment by DATA-xPUNGED** [comment]
> Nitpick: The object icon shows the mask as being pretty thin around the sides, but the onmob has the blue cloth(?) stretched across the whole face like a bandana. Feels inconsistent. (And personally speaking doesn't look captainy)

the side sprite has been fucking with me since i started this, I'll see what i can do

File: code/modules/clothing/masks/gasmask.dm [hunk 1]
```diff
@@ -186,12 +188,22 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
...
+	desc = "A state of the art mask, with a dual purpose slot for either a gas filter or narcotics."
+	actions_types = list(/datum/action/item_action/adjust)
+	flags_inv = HIDEFACIALHAIR | HIDEFACE | HIDESNOUT
+	flags_cover = MASKCOVERSMOUTH | PEPPERPROOF
 	icon_state = "gas_cap"
 	inhand_icon_state = "gasmask_captain"
+	armor_type = /datum/armor/gas_atmos
 	resistance_flags = FIRE_PROOF | ACID_PROOF
+	has_fov = FALSE
+	max_filters = 1
```
Comment (carlarctg) [inline code/modules/clothing/masks/gasmask.dm:202]:
```suggestion
	max_filters = 2
```

Why not two filter slots! One for gas filtering, the other for cigars. Perfectly in character.

Comment (DATA-xPUNGED) [reply code/modules/clothing/masks/gasmask.dm:202]: to put a cigar in, you need to have NO filters on the masks, which is why i removed the extra 2 in the first place

File: code/modules/clothing/masks/gasmask.dm [hunk 2]
```diff
@@ -84,7 +86,7 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
 	var/valid_wearer = ismob(loc)
 	var/mob/wearer = loc
 	if(istype(tool, /obj/item/clothing/mask/cigarette))
-		if(flags_cover & MASKCOVERSMOUTH)
+		if(flags_cover & MASKCOVERSMOUTH & !cig_slot)
```
Comment (Fikou) [inline code/modules/clothing/masks/gasmask.dm:89]:
```suggestion
		if(flags_cover & MASKCOVERSMOUTH && !cig_slot)
```

File: code/modules/clothing/masks/gasmask.dm [hunk 3]
```diff
@@ -186,12 +188,22 @@ GLOBAL_LIST_INIT(clown_mask_options, list(
...
+	flags_cover = MASKCOVERSMOUTH | PEPPERPROOF
 	icon_state = "gas_cap"
 	inhand_icon_state = "gasmask_captain"
+	armor_type = /datum/armor/gas_atmos
 	resistance_flags = FIRE_PROOF | ACID_PROOF
+	has_fov = FALSE
+	max_filters = 1
+	cig_slot = TRUE
+
+/obj/item/clothing/mask/gas/atmos/captain/ui_action_click(mob/user, action)
```
Comment (Fikou) [inline code/modules/clothing/masks/gasmask.dm:205]:
```suggestion
/obj/item/clothing/mask/gas/captain/ui_action_click(mob/user, action)
```

**Review by Fikou — CHANGES_REQUESTED** [review]
it doesnt cover your eyes so it shouldnt make you pepper proof
it should be small and replace the captain's emergency box breathing mask (maybe give him the extended internals tank as well?)

**Comment by Fikou** [comment]
sidesprite has leafy chin

**Comment by Serxule** [comment]
does it protect from pepperspray like the sec gasmask? it should if it doesnt

**Comment by Serxule** [comment]
> it doesnt cover your eyes so it shouldnt make you pepper proof it should be small and replace the captain's emergency box breathing mask (maybe give him the extended internals tank as well?)

sec gasmask protects from pepperspray

**Comment by AtomTheProphet** [comment]
You know, we do have an actual rebreather in the game that's made in xenobiology.

**Comment by DATA-xPUNGED** [comment]
> > it doesnt cover your eyes so it shouldnt make you pepper proof it should be small and replace the captain's emergency box breathing mask (maybe give him the extended internals tank as well?)
> 
> sec gasmask protects from pepperspray

This was removed at some point, by the way

**Comment by Wallemations** [comment]
The object sprite doesn't really look all that great tbh. It could really use some outlines imo

Also what's the difference between this and the normal breath mask people get in their internals boxes??

**Comment by DATA-xPUNGED** [comment]
> The object sprite doesn't really look all that great tbh. It could really use some outlines imo

Very fair tbh, i will try working on something for that

> Also what's the difference between this and the normal breath mask people get in their internals boxes??

This has the atmos gas mask's resistances and armor, on top of being able to put narcotics on the filter slot for captainy purposes

**Comment by DATA-xPUNGED** [comment]
Actually that second point is very fair, i'm gonna try something to make it unique...

**Comment by DATA-xPUNGED** [comment]
Closing this and will open later when i
- Feel like working on this again
- Have a better design for it

**Pull request closed without merging**